test/periodic.jl

```diff
@@ -2,47 +2,32 @@
     @testset "_periodic" begin
         @testset "$f" for f in (sin, cos)
             # A typical 24-hour day
-            for h in 0:24
-                result = _periodic(
-                    f,
-                    ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") + Hour(h),
-                    Day(1),
-                )
-                expected = f(2π / 24 * h)
-                @test result ≈ expected atol=1e-14
-            end
+            x = ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") .+ collect(Hour.(0:2))
+            result = _periodic.(f, x, Day(1))
+            expected = Dict(sin => [0., 0.2588, 0.5], cos => [1.0, 0.9659, 0.866])[f]
```

Comment:
easy to miss the `[f]` at the end here. 

I'd just put `@test result == expected[f] atol=1e-4`